### PR TITLE
Update progress bars

### LINKED ISSUE #970: Progress bar in create_deepcell_output function behaviour
**Describe the bug**
Progress bar in create_deepcell_output function shows incomplete segmentation progress for certain batches. No error msg was generated. Upon inspecting the deepcell output images, the nuclear/whole cell segmentation appears to have been completed.  

**Expected behavior**
Progress bar should show complete if output is generated with no errors

**To Reproduce**
This occurred in a large cohort of 762 images (that was split into 153 batches). The behaviour was consistent on 3 different attempts.  
Screenshot 2023-04-04 at 2 36 42 PM

### LINKED ISSUE #1028: Update progress bar for helpful debugging
**Is your feature request related to a problem? Please describe.**
It would be useful to include the FOV name in any progress bar outputs, so that when something goes wrong during processing the user can know exactly at which point the problem occurred.

**Describe the solution you'd like**
1. Implement progress bars for any major processing functions that do not have them.
   - fiber segmentation
2. Update any existing progress bars to have the FOV name in the output.
3. For functions that already include the FOV name, make sure to update the name at the beginning of processing and not the end.
   - [Currently](https://github.com/angelolab/ark-analysis/blob/b09107e3de49a3acf3177e418a5dde0da011f76f/src/ark/analysis/spatial_analysis_utils.py#L69), the FOV name and the completion status are updated once processing is complete; this results in the previous FOV name still displayed when the function errors out.

**Describe alternatives you've considered**
An atrocious amount of print statements to let the user know which FOV is being processed.

**Additional context**
We can also implement this progress bar structure in toffy as well (most useful for longer processing such as rosetta and normalization).

### PR DESCRIPTION
**Pull request by camisowers**
**If you haven't already, please read through our [contributing guidelines](https://ark-analysis.readthedocs.io/en/latest/_rtd/contributing.html) before opening your PR**

**What is the purpose of this PR?**

1. Closes #1028. Add current processing FOV name to progress bar output in the ark notebooks.
2. Closes #970. Fix incorrect progress bars that show <100% even though the batch was processed fine.

**How did you implement your changes**

1. Update the FOV name at start of each code block using `progress_bar.set_postfix(FOV=fov)`.

2. I was able to replicate @HPiyadasa's issue, and it turns out two things could potentially happen with the current code:
- Since we only check the progress every 3 seconds, sometimes we don't always know that the last fov is in progress before getting a "done" response from the api and closing the progress bar. We simply need to ensure the progress bar gets updated for completion before breaking the batch loop and closing the progress bar.

Screenshot 2023-08-01 at 1 58 49 PM (2)

- (This one is quirky) Sometimes deepcell.org/api/redis will return a lower completion rate than it had before.  When combined with the above issue, the math doesn't math to 100. We just need to account for this by only updating the progress bar status when it has increased.

Screenshot 2023-08-01 at 12 35 06 PM (2)

**Remaining issues**

Nothing, I've solved all issues forever.

Screenshot 2023-08-01 at 2 27 44 PM (2)

### CODE REVIEW

File: src/ark/segmentation/fiber_segmentation.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 from skimage.measure import regionprops_table
 from skimage.morphology import remove_small_objects
 from skimage.segmentation import watershed
+from tqdm.notebook import tqdm
```
Comment (srivarra) [inline src/ark/segmentation/fiber_segmentation.py:19]: we should use `from tqdm.auto import tqdm` so `tqdm` can decide where it's running.

File: src/ark/analysis/neighborhood_analysis.py [hunk 2]
```diff
@@ -79,6 +79,8 @@ def create_neighborhood_matrix(all_data, dist_mat_dir, included_fovs=None, distl
     with tqdm(total=len(included_fovs), desc="Neighbors Matrix Generation") \
```
Comment (srivarra) [inline src/ark/analysis/neighborhood_analysis.py:79]: `unit` here

File: src/ark/analysis/spatial_analysis_utils.py [hunk 3]
```diff
@@ -39,6 +39,8 @@ def calc_dist_matrix(label_dir, save_path, prefix='_whole_cell'):
     # iterate for each fov
     with tqdm(total=len(fov_files), desc="Distance Matrix Generation") as dist_mat_progress:
```
Comment (srivarra) [inline src/ark/analysis/spatial_analysis_utils.py:40]: `unit` here

File: src/ark/analysis/spatial_enrichment.py [hunk 4]
```diff
@@ -72,6 +72,8 @@ def generate_channel_spatial_enrichment_stats(label_dir, dist_mat_dir, marker_th
 
     with tqdm(total=len(all_label_fovs), desc="Channel Spatial Enrichment") as chan_progress:
```
Comment (srivarra) [inline src/ark/analysis/spatial_enrichment.py:73]: `unit` here

File: src/ark/segmentation/fiber_segmentation.py [hunk 5]
```diff
@@ -176,14 +177,20 @@ def run_fiber_segmentation(data_dir, fiber_channel, out_dir, img_sub_folder=None
...
 
-    for fov in fovs:
-        print(f'Processing FOV: {fov}')
-        subset_xr = load_utils.load_imgs_from_tree(
-            data_dir, img_sub_folder, fovs=fov, channels=[fiber_channel]
-        )
-        subtable = segment_fibers(subset_xr, fiber_channel, out_dir, fov, save_csv=False,
-                                  **kwargs)
-        fiber_object_table.append(subtable)
+    with tqdm(total=len(fovs), desc="Fiber Segmentation") as fibseg_progress:
```
Comment (srivarra) [inline src/ark/segmentation/fiber_segmentation.py:180]: `unit` here

File: src/ark/utils/data_utils.py [hunk 6]
```diff
@@ -463,6 +465,8 @@ def generate_and_save_pixel_cluster_masks(fovs: List[str],
     # create the pixel cluster masks across each fov
     with tqdm(total=len(fovs), desc="Pixel Cluster Mask Generation") as pixel_mask_progress:
         for fov in fovs:
```
Comment (srivarra) [inline src/ark/utils/data_utils.py:467]: `unit` here.

File: src/ark/utils/data_utils.py [hunk 7]
```diff
@@ -537,6 +541,8 @@ def generate_and_save_neighborhood_cluster_masks(
     ) as neigh_mask_progress:
         # generate the mask for each FOV
         for fov in fovs:
+            neigh_mask_progress.set_postfix(FOV=fov)
```
Comment (srivarra) [inline src/ark/utils/data_utils.py:544]: `unit`  here.

**Review by srivarra — CHANGES_REQUESTED** [review]
Looks great, just a `tqdm` import suggestion and some units are missing in a few loops

**Review by alex-l-kong — COMMENTED** [review]
Nothing to add on top of Sri's comments.

**Review by srivarra — APPROVED** [review]
Looks good!

**Review by alex-l-kong — APPROVED** [review]

**Comment by HPiyadasa** [comment]
Looks great

**Review by HPiyadasa — APPROVED** [review]

**Pull request merged**